src/com/xilinx/rapidwright/edif/EDIFPortInstList.java

```diff
@@ -114,4 +116,19 @@ private int compare(EDIFPortInst left, String rightInstName, String rightPortIns
         int compare = left.getCellInst().getName().compareTo(rightInstName);
         return compare == 0 ? left.getName().compareTo(rightPortInstName) : compare;
     }
+
+    public void _parseAdd(EDIFPortInst e) {
+        super.add(e);
+    }
+
+    public void _reSortList() {
+        Collections.sort(this, new Comparator<EDIFPortInst>() {
+            public int compare(EDIFPortInst left, EDIFPortInst right) {
+                EDIFCellInst cellInst = right.getCellInst();
+                String cellInstName = cellInst == null ? null : cellInst.getName();
+                String portInstName = right.getName();
+                return EDIFPortInstList.compare(left, cellInstName, portInstName);
```

Comment:
Might it be better to have `EDIFPortInst` implement the [`Comparator`](https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html) interface directly?